tensorflow-core/tensorflow-core-api/src/main/java/org/tensorflow/EagerOperationBuilder.java

```diff
@@ -60,23 +58,28 @@
     this.session = session;
     this.type = type;
     this.name = name;
-    this.nativeRef = new NativeReference(session, this, allocate(session.nativeHandle(), type));
+    this.opHandle = allocate(session.nativeHandle(), type);
+    session.attach(opHandle);
+    opHandle.releaseReference();
```

Comment:
If I recall correctly what @EronWright was envisioning for the reference counts is that it could be possible for the `session` to "grab" the ownership of the `opHandle` when it gets attached to it, instead of explicitly releasing it like in few places in this PR. 

I guess it is done like this because in JavaCPP, you need to explicitly retain a reference so its count is `1`, instead of having an initial value of `1`. Would you mind explaining again why it needs to be like this?

Reply:
When we're inside a scope (as in `PointerScope`), the count does get initialized to 1 by default. But when we're outside of any scope, it stays to 0, as a flag to indicate that reference counting is disabled. We need to enable it manually. If a `Pointer` would own itself automatically, then the count would start at `2` when initialized inside a scope, and so users would need to release them manually, even inside of a scope. It would make the use of scope pointless.

The idea is to do everything with `PointerScope`, then we don't need to call anything else, and everything is done behind the scenes, since the scopes own all the `Pointer` objects. It works exactly like RAII in C++. If you would like to understand better the motivation behind this, please read up on C++ and RAII!

Reply:
Then why not working to integrate a bit better the `PointerScope` in this PR? How would that work, the `EagerSession` needs to extend from it?

Reply:
No, we just need to change a few things to be able to pass scopes around a bit more. Sure, let's do it now while we're at it.

Reply:
Ok, it's done!